src/app/auth/login/login.service.ts

```diff
@@ -1,21 +1,46 @@
 import { Injectable } from "@angular/core";
-import { Observable, Subject, map } from "rxjs";
+import { Observable, ReplaySubject, Subject, map } from "rxjs";
 import { AuthApi } from "src/app/api/auth.api";
 import { GithubLoginCredentials, PasswordLoginCredentials } from "src/app/api/auth.api.model";
+import { AccountFragment, LoginResponse } from "src/app/api/kamu.graphql.interface";
 import { LoginMethod } from "src/app/app-config.model";
 import { AuthenticationError } from "src/app/common/errors";
 import ProjectLinks from "src/app/project-links";
 import { NavigationService } from "src/app/services/navigation.service";
+import { LoginCallbackResponse } from "./login.component.model";
+import { HttpClient } from "@angular/common/http";
+import { AppConfigService } from "src/app/app-config.service";
+import { MaybeNull } from "src/app/common/app.types";
+import { LocalStorageService } from "src/app/services/local-storage.service";
 
 @Injectable({
     providedIn: "root",
 })
 export class LoginService {
-    public constructor(private authApi: AuthApi, private navigationService: NavigationService) {}
+    public constructor(
+        private authApi: AuthApi,
+        private navigationService: NavigationService,
+        private appConfigService: AppConfigService,
+        private localStorageService: LocalStorageService,
+        private httpClient: HttpClient,
+    ) {}
+
+    private accessTokenObtained$: Subject<string> = new ReplaySubject<string>(1);
+    private accountChanged$: Subject<AccountFragment> = new ReplaySubject<AccountFragment>(1);
 
     private errorPasswordLogin$: Subject<string> = new Subject<string>();
     private enabledLoginMethods: LoginMethod[] = [];
 
+    private loginCallback: (loginResponse: LoginResponse) => void = this.redirectUrlLoginCallback.bind(this);
+
+    public accessTokenObtained(): Observable<string> {
+        return this.accessTokenObtained$.asObservable();
+    }
+
+    public accountChanged(): Observable<AccountFragment> {
+        return this.accountChanged$.asObservable();
+    }
```

Comment:
For meaningful naming, I suggest changing methods to properties or renaming methods

Properties:
public accessToken$ = this.accessTokenObtained$.asObservable();
public account$ = this.accountChanged$.asObservable();

or

Methods:
getAccessTokenObservable
getAccountObservable

Reply:
I agree with the arguments, but I want to mention that the first option wasn't about exposing a subject,
the subject remains private, but the method changes to a property and exposes observable.

But there are no strong points about it, I saw both variants I assume it's more a matter of taste